### DISCUSSION TITLE
Filter DNS server sources based on category/what they block.

### DISCUSSION THREAD

**Discussion by 2brownc** [DNSCrypt/dnscrypt-proxy#2065] [Ideas]
There is already a filter `require_nofilter` that ensures the servers selected won't enforce their own blacklist. 

But I propose filter that SELECTS only server from a given category. 

Eg:

DNS Server A - block ads, gambling, malware
DNS Server B - gambling, malware, cryptominers
DNS Server C - block ads, privacy, fakenews
DNS Server D - block ads, malware

Now in the configuration file:

```
require_blockCategory = ['ads', 'malware']
```

Now only servers that block ads and malware are selects .i.e servers A and D.

PS: Already a [list of DNS servers](https://github.com/dnscrypt/dnscrypt-resolvers) and what they block is being maintained.

**Comment by TehAppKiller** [comment]
Up !

Today, we can only have random-filtering or no filtering at all.
Random filtering as servers can have ads, malware, family.

Yes there are blacklist possible, and custom resolvers lists but it
Would be good to add a filter that at least match server description criteria.
e.g. :
filter = ['ads'] // Block only ads if 'ads' is in resolver description
filter = ['ads', 'malware'] // Block ads & malware if 'ads' & 'malware' are in resolver description
...

Has to be careful with load-balancing with numbers of servers (for 'ph' for example) I guess
In 2nd time, it would be good to be a bitmask (Malware=1, No-Ads=2, Family=4, ...)

Thx <3
